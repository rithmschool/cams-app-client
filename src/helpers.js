export const BASE_URL = process.env.REACT_APP_SERVER_URL || 'http://localhost:5000'

export const ensureCorrectUser = (pathId) => {
  let token = localStorage.getItem('token');
  if(!token){return false;}
  let tokenId = JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id
  return +tokenId === +pathId;
}

export const userID = () => JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id

export const config = {
  headers: {
    'Accept':'application/json',
    'ContentType':'application/json',
    'Authorization':'bearer ' + localStorage.getItem('token')
  }
}
