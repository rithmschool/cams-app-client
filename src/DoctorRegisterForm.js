import React, { Component } from "react";
import { BASE_URL, config } from "./helpers.js";
import axios from "axios";

class DoctorRegisterForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      password: "",
      confirmPassword: "",
      response: "",
      displayForm: false
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  componentWillMount() {
    var token = this.props.location.search.split("=")[1];
    axios
      .post(`${BASE_URL}/api/users/register`, {
        token: token
      })
      .then(response => {
        this.setState({
          response: response.data.message,
          displayForm: true
        });
      })
      .catch(response => {
        this.props.history.push(`/`);
      });
  }

  registerDoctor(config) {
    var token = this.props.location.search.split("=")[1];
    axios
      .patch(`${BASE_URL}/api/users/register`, {
        token: token,
        password: this.state.password,
        confirmPassword: this.state.confirmPassword
      })
      .then(response => {
        this.setState({
          response: response.data.message
        });
        setTimeout(() => {
          this.props.history.push(`/login`);
        }, 3000);
      })
      .catch(response => {
        this.setState({
          response: response.response.data.message
        });
      });
    this.setState({
      response: "Setting up your account, one moment..."
    });
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  handleSubmit(e) {
    e.preventDefault();
    if (!this.state.password || !this.state.confirmPassword) {
      this.setState({
        response: "Cannot have an empty password."
      });
      return;
    }
    this.registerDoctor(config, this);
    this.setState({
      password: "",
      confirmPassword: ""
    });
  }

  render() {
    let responseText = <p>{this.state.response}</p>;
    let passwordForm = (
      <div>
        <h1>Create New Password</h1>
        <form onSubmit={this.handleSubmit}>
          <input
            type="password"
            name="password"
            placeholder="new password"
            value={this.state.password}
            onChange={this.handleChange}
          />
          <input
            type="password"
            name="confirmPassword"
            placeholder="confirm new password"
            value={this.state.confirmPassword}
            onChange={this.handleChange}
          />
          <button className="button button-hover" type="submit" value="Submit">
            Submit
          </button>
        </form>
      </div>
    );
    return (
      <div>
        <div className="banner-text">
          <h1 className="banner-bold">Register</h1>
        </div>
        <div className="content">
          {this.state.displayForm ? passwordForm : null}
          {responseText}
        </div>
      </div>
    );
  }
}

export default DoctorRegisterForm;
