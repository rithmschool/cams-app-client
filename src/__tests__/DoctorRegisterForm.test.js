import DoctorRegisterForm from "../DoctorRegisterForm";
import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LocalStorageMock from "../setupTests/LocalStorageMock";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import { MemoryRouter } from "react-router-dom";
import { BASE_URL } from "../helpers.js";

var mock = new MockAdapter(axios);

beforeAll(() => {
  localStorage.setItem("token", ".eyJpZCI6NH0=.");
  mock
    .onPost(`${BASE_URL}/api/users/register`)
    .reply(200, { token: "token", response: "something", displayForm: false });
});

test("DoctorRegisterForm snapshot", () => {
  const doctorRegisterForm = shallow(
    <DoctorRegisterForm
      location={{
        pathname: "/doctor/register",
        search: "?token=InNAc2FyYWguc295Ig.DL11ag.rqelLIUl2I8C33G_y7dc8ysJVsE",
        hash: "",
        state: undefined
      }}
    />
  );
  doctorRegisterForm.setState({
    password: "",
    confirmPassword: "",
    response: "",
    displayForm: false
  });
  expect(toJson(doctorRegisterForm)).toMatchSnapshot();
});

afterAll(() => {
  mock.restore();
});
