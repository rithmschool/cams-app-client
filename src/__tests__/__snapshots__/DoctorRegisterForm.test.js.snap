// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorRegisterForm snapshot 1`] = `
<div>
  <div
    className="banner-text"
  >
    <h1
      className="banner-bold"
    >
      Register
    </h1>
  </div>
  <div
    className="content"
  >
    <p />
  </div>
</div>
`;
